src/analysis/inner_product_space/lax_milgram.lean

```diff
@@ -0,0 +1,142 @@
+variables {B : V →L[ℝ] V →L[ℝ] ℝ}
+
+local postfix `♯`:1025 := @continuous_linear_map_of_bilin ℝ V _ _ _
+
+lemma bounded_below (coercive : is_coercive B) :
+  ∃ C, 0 < C ∧ ∀ v, C * ∥v∥ ≤ ∥B♯ v∥ :=
+begin
+  rcases coercive with ⟨C, C_ge_0, coercivity⟩,
+  refine ⟨C, C_ge_0, _⟩,
+  intro v,
+  by_cases h : 0 < ∥v∥,
+  { refine (mul_le_mul_right h).mp _,
+    exact calc C * ∥v∥ * ∥v∥
+               ≤ B v v                         : coercivity v
+    ...        = inner (B♯ v) v : by simp
+    ...        ≤ ∥B♯ v∥ * ∥v∥     : real_inner_le_norm (B♯ v) v, },
+  { have : v = 0 := by simpa using h,
+    simp [this], }
+end
+
+lemma antilipschitz_of_lax_milgram (coercive : is_coercive B) :
+  ∃ C : nnreal, 0 < C ∧ antilipschitz_with C B♯ :=
+begin
+  rcases bounded_below coercive with ⟨C, C_pos, below_bound⟩,
+  refine ⟨(C⁻¹).to_nnreal, real.to_nnreal_pos.mpr (inv_pos.mpr C_pos), _⟩,
+  refine linear_map.antilipschitz_of_bound B♯ _,
+  simp_rw [real.coe_to_nnreal',
+    max_eq_left_of_lt (inv_pos.mpr C_pos),
+    ←inv_mul_le_iff (inv_pos.mpr C_pos)],
+  simpa using below_bound,
+end
+
+lemma ker_eq_bot (coercive : is_coercive B) : B♯.ker = ⊥ :=
+begin
+  rw [←ker_coe, linear_map.ker_eq_bot],
+  rcases antilipschitz_of_lax_milgram coercive with ⟨_, _, antilipschitz⟩,
+  exact antilipschitz_with.injective antilipschitz,
+end
+
+lemma closed_range (coercive : is_coercive B) : is_closed (B♯.range : set V) :=
+begin
+  rcases antilipschitz_of_lax_milgram coercive with ⟨_, _, antilipschitz⟩,
+  exact antilipschitz.is_closed_range B♯.uniform_continuous,
+end
+
+lemma range_eq_top (coercive : is_coercive B): B♯.range = ⊤ :=
+begin
+  haveI := (closed_range coercive).complete_space_coe,
+  rw ← B♯.range.orthogonal_orthogonal,
+  rw submodule.eq_top_iff',
+  intros v w mem_w_orthogonal,
+  rcases coercive with ⟨C, C_ge_0, coercivity⟩,
+  have : C * ∥w∥ * ∥w∥ ≤ 0 :=
+  calc C * ∥w∥ * ∥w∥
+        ≤ B w w                         : coercivity w
+  ...  = inner (B♯ w) w   : by simp
+  ...  = 0                              : mem_w_orthogonal _ ⟨w, rfl⟩,
+  have : ∥w∥ * ∥w∥ ≤ 0 := by nlinarith,
+  have h : ∥w∥ = 0 := by nlinarith [norm_nonneg w],
+  have w_eq_zero : w = 0 := by simpa using h,
+  simp [w_eq_zero],
+end
+
+/--
+The Lax-Milgram equivalence of a coercive bounded bilinear operator:
+for all `v : V`, `lax_milgram_equiv B v` is the unique element `V`
+such that `⟪lax_milgram_equiv B v, w⟫ = B v w`.
+The Lax-Milgram theorem states that this is a continuous equivalence.
+-/
+def continuous_linear_equiv_of_bilin (coercive : is_coercive B) : V ≃L[ℝ] V :=
+continuous_linear_equiv.of_bijective
+  (B♯)
+  (ker_eq_bot coercive)
+  (range_eq_top coercive)
+
+
+@[simp]
+lemma continuous_linear_equiv_of_bilin_apply (coercive : is_coercive B) (v w : V) :
+  inner (continuous_linear_equiv_of_bilin coercive v) w = B v w :=
+continuous_linear_map_of_bilin_apply B v w
+
+lemma continuous_linear_equiv_of_bilin_equiv (coercive : is_coercive B) {v f : V}
```

Comment:
Would it make sense to call this lemma `unique_continuous_linear_equiv_of_bilin`?

Reply:
Yes, that's what the correct name should be.